code/iaas/simple-allocator/src/main/java/io/cattle/platform/simple/allocator/SimpleAllocator.java

```diff
@@ -453,24 +456,30 @@ private ResourceRequest populateResourceRequestFromInstance(Instance instance, S
 
     @Override
     protected boolean useLegacyPortAllocation(Long accountId, List<Instance> instances) {
-        Long agentId = getAgentResource(accountId, instances);
-        return useLegacyPortAllocation(agentId);
-    }
-
-    protected boolean useLegacyPortAllocation(Long agentId) {
-        if (agentId == null) {
-            // No scheduler, use legacy logic
+        List<Long> agentIds = getAgentResource(accountId, instances);
+        if (agentIds == null || agentIds.size() == 0) {
             return true;
         }
+        return useLegacyPortAllocation(agentIds);
+    }
 
-        String schedulerVersion = getSchedulerVersion(agentId);
-        return useLegacyPortAllocation(schedulerVersion);
+    protected boolean useLegacyPortAllocation(List<Long> agentIds) {
+        for(Long agentId: agentIds) {
+            String schedulerVersion = getSchedulerVersion(agentId);
+            if (useLegacyPortAllocation(schedulerVersion)) {
+                return true;
+            }
+        }
+        return false;
     }
 
     protected String getSchedulerVersion(Long agentId) {
         Instance instance = agentInstanceDao.getInstanceByAgent(agentId);
         String imageUuid = (String) DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_IMAGE_UUID).get();
         DockerImage img = DockerImage.parse(imageUuid);
+        if (!StringUtils.startsWith(img.getFullName(), "rancher/scheduler")) {
```

Comment:
check startwith("rancher/scheduler:") or equal("rancher/scheduler")? Is that sufficient?